<a href="https://colab.research.google.com/github/abhishek-parashar/Reddit-flair-detection/blob/master/experiment_log_for_eda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### importing the required libraries

In [44]:
import numpy as np 
import pandas as pd 
import seaborn as sns
import matplotlib.pyplot as plt
import numpy as np 
import pandas as pd
import re
import nltk
from nltk.corpus import stopwords
import datetime as dt
nltk.download('all')
from bs4 import BeautifulSoup

[nltk_data] Downloading collection 'all'
[nltk_data]    | 
[nltk_data]    | Downloading package abc to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/abc.zip.
[nltk_data]    | Downloading package alpino to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/alpino.zip.
[nltk_data]    | Downloading package biocreative_ppi to
[nltk_data]    |     /root/nltk_data...
[nltk_data]    |   Unzipping corpora/biocreative_ppi.zip.
[nltk_data]    | Downloading package brown to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/brown.zip.
[nltk_data]    | Downloading package brown_tei to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/brown_tei.zip.
[nltk_data]    | Downloading package cess_cat to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/cess_cat.zip.
[nltk_data]    | Downloading package cess_esp to /root/nltk_data...
[nltk_data]    |   Unzipping corpora/cess_esp.zip.
[nltk_data]    | Downloading package chat80 to /root/nltk_data...
[nltk_data]    |   Unzipp

## reading the file 


In [30]:
pd.read_csv('data.csv')

sno.  ...                                  combined_features
0        0  ...  4 days ago pending orders 100 million hydroxyc...
1        1  ...  randians big time users dating apps like tinde...
2        2  ...  r india thinks flat earthershavent found india...
3        3  ...  people left 9 5 jobs pursue career music art f...
4        4  ...  somebody want kill full familycalm downgo sp o...
...    ...  ...                                                ...
1245  1213  ...  coronavirus india hunger real problemnanhttps:...
1246  1214  ...  need investigate nso group behind pegasus soft...
1247  1215  ...  coronavirus dont target doctors asking protect...
1248  1216  ...  india lifts export ban potential coronavirus d...
1249  1217  ...  33 #covid19 positive cases reported haryana to...

[1250 rows x 12 columns]

In [31]:
data.head()

flair  ...                                  combined_features
0  AskIndia  ...  4 days ago pending orders 100 million hydroxyc...
1  AskIndia  ...  randians big time users dating apps like tinde...
2  AskIndia  ...  r india thinks flat earthershavent found india...
3  AskIndia  ...  people left 9 5 jobs pursue career music art f...
4  AskIndia  ...  somebody want kill full familycalm downgo sp o...

[5 rows x 11 columns]

looking for the data shape 

In [25]:
data.shape

(267, 11)

names of columns

In [32]:
data.columns

Index(['flair', 'title', 'score', 'id', 'url', 'comms_num', 'body', 'author',
       'comments', 'timestamp', 'combined_features'],
      dtype='object')

In [35]:
data.head()

flair  ...                                  combined_features
0  AskIndia  ...  4 days ago pending orders 100 million hydroxyc...
1  AskIndia  ...  randians big time users dating apps like tinde...
2  AskIndia  ...  r india thinks flat earthershavent found india...
3  AskIndia  ...  people left 9 5 jobs pursue career music art f...
4  AskIndia  ...  somebody want kill full familycalm downgo sp o...

[5 rows x 11 columns]

checking the data info

In [36]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 267 entries, 0 to 266
Data columns (total 11 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   flair              266 non-null    object 
 1   title              266 non-null    object 
 2   score              266 non-null    float64
 3   id                 266 non-null    object 
 4   url                266 non-null    object 
 5   comms_num          266 non-null    float64
 6   body               182 non-null    object 
 7   author             266 non-null    object 
 8   comments           249 non-null    object 
 9   timestamp          266 non-null    object 
 10  combined_features  266 non-null    object 
dtypes: float64(2), object(9)
memory usage: 23.1+ KB


we are getting only 266 values whereas the data has1250 rows

In [0]:
data.drop("id",axis=1,inplace=True)

In [38]:
data.head()

flair  ...                                  combined_features
0  AskIndia  ...  4 days ago pending orders 100 million hydroxyc...
1  AskIndia  ...  randians big time users dating apps like tinde...
2  AskIndia  ...  r india thinks flat earthershavent found india...
3  AskIndia  ...  people left 9 5 jobs pursue career music art f...
4  AskIndia  ...  somebody want kill full familycalm downgo sp o...

[5 rows x 10 columns]

In [39]:
data.columns

Index(['flair', 'title', 'score', 'url', 'comms_num', 'body', 'author',
       'comments', 'timestamp', 'combined_features'],
      dtype='object')

In [40]:
data.tail()

flair  ...                                  combined_features
262  Scheduled  ...  hi r india made bot stop fake news whatsappmak...
263  Scheduled  ...  get full refund jet airways international flig...
264  Scheduled  ...  seattle city council becomes first us legislat...
265  Scheduled  ...  confusion train schedule could someone please ...
266  Scheduled  ...  punjab dalit labourer beaten forced drink urin...

[5 rows x 10 columns]

In [0]:
def string(value):
    return str(value)

In [0]:
data['title'] = data['title'].apply(string)
data['body'] = data['body'].apply(string)
data['comments'] = data['comments'].apply(string)

In [0]:

REPLACE_BY_SPACE_RE = re.compile('[/(){}\[\]\|@,;]')
BAD_SYMBOLS_RE = re.compile('[^0-9a-z #+_]')
STOPWORDS = set(stopwords.words('english'))
def text_cleaning(text):
   
    text = BeautifulSoup(text, "lxml").text
    text = text.lower()
    text = REPLACE_BY_SPACE_RE.sub(' ', text)
    text = BAD_SYMBOLS_RE.sub('', text)
    text = ' '.join(word for word in text.split() if word not in STOPWORDS)
    return text

In [0]:
combined_features = data["title"] + data["comments"] + data["url"]
data = data.assign(combined_features = combined_features)

In [47]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 267 entries, 0 to 266
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   flair              266 non-null    object 
 1   title              267 non-null    object 
 2   score              266 non-null    float64
 3   url                266 non-null    object 
 4   comms_num          266 non-null    float64
 5   body               267 non-null    object 
 6   author             266 non-null    object 
 7   comments           267 non-null    object 
 8   timestamp          266 non-null    object 
 9   combined_features  266 non-null    object 
dtypes: float64(2), object(8)
memory usage: 21.0+ KB


In [0]:
data1=pd.read_csv('data (1).csv')

In [50]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1218 entries, 0 to 1217
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Unnamed: 0         1218 non-null   int64 
 1   flair              1218 non-null   object
 2   title              1218 non-null   object
 3   score              1218 non-null   int64 
 4   id                 1218 non-null   object
 5   url                1218 non-null   object
 6   comms_num          1218 non-null   int64 
 7   body               693 non-null    object
 8   author             1218 non-null   object
 9   comments           1096 non-null   object
 10  timestamp          1218 non-null   object
 11  combined_features  1218 non-null   object
dtypes: int64(3), object(9)
memory usage: 114.3+ KB


### by this we find that body has a lot of null values. This shows that for most of the posts only the title suffice, there is no need of body. 
### plots can be made for the posts having body and no of words in such posts this would give more insights to the data. 
### second in the list which has missing values is comments. This is quite resonable as not all the posts get comments, next thing which can be checked is what is the score of the post with comments and what flair are the posts which get maximum comments. 


#### graphs which we can plot to get more insights of the data 
#### 1. plots of different feature values and which feature value has missing values.
#### 2. NO of posts and the flair they belong that have body.
#### 3. No of words in the body corresponding to the flairs- this will help us determine is body a good parameter or not for model.
#### 4. score of the posts with comments
#### 5. flair which gets maximum comments - this will help us in model 
#### 6. what are thw words or persons mentioned maximum times in comments
#### 7, what is the frequency of posts and comments.

AttributeError: ignored